packages/core/src/utils/geminiIgnoreParser.ts

```diff
@@ -0,0 +1,81 @@
+/**
+ * @license
+ * Copyright 2025 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as fs from 'node:fs';
+import * as path from 'node:path';
+import ignore from 'ignore';
+
+export interface GeminiIgnoreFilter {
+  isIgnored(filePath: string): boolean;
+  getPatterns(): string[];
+}
+
+export class GeminiIgnoreParser implements GeminiIgnoreFilter {
+  private projectRoot: string;
+  private patterns: string[] = [];
+  private ig = ignore();
+
+  constructor(projectRoot: string) {
+    this.projectRoot = path.resolve(projectRoot);
+    this.loadPatterns();
+  }
+
+  private loadPatterns(): void {
+    const patternsFilePath = path.join(this.projectRoot, '.geminiignore');
+    let content: string;
+    try {
+      content = fs.readFileSync(patternsFilePath, 'utf-8');
```

Comment:
this is probably the right tradeoff but a little concerned the file reads are now sync.  could be worth to consider what it would take to make this non-blocking. it might cause too many ripple effect changes to be worth it.